src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -2850,4 +2850,409 @@ public class RequiredMemberAttribute : Attribute
+        Prop3.ToString(); // 2
+        Prop4.ToString();
+    }}
+}}
+";
+
+        var comp = CreateCompilationWithRequiredMembers(code);
+        comp.VerifyDiagnostics(
+            // (16,9): warning CS8602: Dereference of a possibly null reference.
+            //         Prop1.ToString(); // 1
+            Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "Prop1").WithLocation(16, 9),
+            // (18,9): warning CS8602: Dereference of a possibly null reference.
+            //         Prop3.ToString(); // 2
+            Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "Prop3").WithLocation(18, 9)
+        );
+    }
+
+    [Theory, CompilerTrait(CompilerFeature.NullableReferenceTypes)]
+    [InlineData(": base()")]
+    [InlineData("")]
+    public void RequiredMemberSuppressesNullabilityWarnings_ChainedConstructor_06(string baseSyntax)
+    {
+        var code = @$"
+#nullable enable
+public class Base
+{{
+    public required string Prop1 {{ get; set; }}
+    public string Prop2 {{ get; set; }} = null!;
+}}
+
+public class Derived : Base
+{{
+    public required string Prop3 {{ get; set; }}
+    public string Prop4 {{ get; set; }}
+
+    public Derived() {baseSyntax}
+    {{
+        Prop1.ToString(); // 1
+        Prop2.ToString();
+        Prop3.ToString(); // 2
+        Prop4.ToString(); // 3
+    }}
+}}
+";
+
+        var comp = CreateCompilationWithRequiredMembers(code);
+        comp.VerifyDiagnostics(
+            // (16,9): warning CS8602: Dereference of a possibly null reference.
+            //         Prop1.ToString(); // 1
+            Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "Prop1").WithLocation(16, 9),
+            // (18,9): warning CS8602: Dereference of a possibly null reference.
+            //         Prop3.ToString(); // 2
+            Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "Prop3").WithLocation(18, 9),
+            // (19,9): warning CS8602: Dereference of a possibly null reference.
+            //         Prop4.ToString(); // 3
+            Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "Prop4").WithLocation(19, 9)
+        );
+    }
+
+    [Fact, CompilerTrait(CompilerFeature.NullableReferenceTypes)]
+    public void RequiredMemberSuppressesNullabilityWarnings_ChainedConstructor_07()
+    {
+        var code = @"
+#nullable enable
+public class Base
+{
+    public required string Prop1 { get; set; }
+    public string Prop2 { get; set; } = null!;
+}
+
+public class Derived : Base
+{
+    public required string Prop3 { get; set; }
+    public string Prop4 { get; set; }
+
+    public Derived(bool unused) { Prop4 = null!; }
+
+    public Derived() : this(true)
+    {
+        Prop1.ToString(); // 1
+        Prop2.ToString();
+        Prop3.ToString(); // 2
+        Prop4.ToString();
+    }
+}
+";
+
+        var comp = CreateCompilationWithRequiredMembers(code);
+        comp.VerifyDiagnostics(
+            // (18,9): warning CS8602: Dereference of a possibly null reference.
```

Comment:
nit: All the tests about CS8602 use properties. Consider swapping one of the tests to use fields.